sdk/monitor/Azure.Monitor.Query/src/LogsClient.cs

```diff
@@ -87,15 +89,80 @@ public virtual LogsBatchQuery CreateBatchQuery()
             return new LogsBatchQuery(_clientDiagnostics, _queryClient, _rowBinder);
         }
 
-        internal static QueryBody CreateQueryBody(string query, TimeSpan? timeSpan)
+        internal static QueryBody CreateQueryBody(string query, TimeSpan? timeSpan, LogsQueryOptions options, out string prefer)
         {
             var queryBody = new QueryBody(query);
             if (timeSpan != null)
             {
                 queryBody.Timespan = TypeFormatters.ToString(timeSpan.Value, "P");
             }
 
+            StringBuilder preferBuilder = null;
+            if (options?.Timeout is TimeSpan timeout)
```

Comment:
What else can `options.Timeout` be?

Reply:
`null`

Reply:
Why not `HasValue` rather than type checking?

Reply:
Shorter usage `timeout` vs `options?.Timeout.Value`

but also being cool.

![](https://thumbs.gfycat.com/BeautifulAffectionateLarva-max-1mb.gif)

Reply:
I kind of like this pattern of narrowing down a value and giving it an alias.